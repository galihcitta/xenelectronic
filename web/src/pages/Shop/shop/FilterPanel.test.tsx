import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { FilterPanelProps, _FilterPanel } from './FilterPanel.component';

describe('<FilterPanel />', () => {
  const mockSetGridColumns = jest.fn();
  const mockFilterProducts = jest.fn();

  const props: FilterPanelProps = {
    colorFilters: [
      {
        description: 'black',
        amountOfProducts: 2
      },
      {
        description: 'white',
        amountOfProducts: 3
      }
    ],
    characterFilters: [
      {
        description: 'modern',
        amountOfProducts: 2
      },
      {
        description: 'classic',
        amountOfProducts: 3
      }
    ],
    fetchedProductsAmount: 5,
    minPrice: 100,
    maxPrice: 300,
    query: '',
    gridColumns: 2,
    setGridColumns: mockSetGridColumns,
    filterProducts: mockFilterProducts
  };

  const setup = (props: FilterPanelProps) => {
    const utils = render(<_FilterPanel {...props} />);
    const gridOptionsBtns = utils.container.querySelectorAll('button');

    return { ...utils, gridOptionsBtns };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('grid options buttons', () => {
    it('contains correct style rule', () => {
      const { gridOptionsBtns } = setup(props);

      expect(gridOptionsBtns[0]).toHaveStyle('color: #151615');
      expect(gridOptionsBtns[1]).toHaveStyle('color: #bbb');
    });

    it('calls `setGridColumns` func', () => {
      const { gridOptionsBtns } = setup(props);

      fireEvent.click(gridOptionsBtns[0]);
      expect(mockSetGridColumns).toHaveBeenCalledTimes(1);
      expect(mockSetGridColumns).toBeCalledWith(2);

      fireEvent.click(gridOptionsBtns[1]);
      expect(mockSetGridColumns).toHaveBeenCalledTimes(2);
      expect(mockSetGridColumns).toBeCalledWith(3);
    });
  });

  it('contains `QueryResult` element with query value as text', () => {
    const { getByText } = setup({ ...props, query: 'clocks' });
    const queryResult = getByText(/search result/i);

    expect(queryResult).toBeInTheDocument();
    expect(queryResult).toHaveTextContent('clocks');
  });
});
