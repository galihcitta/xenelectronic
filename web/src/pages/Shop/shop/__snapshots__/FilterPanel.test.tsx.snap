// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPanel /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dpDXXu"
  >
    <div
      class="sc-ifAKCX bUenFB"
    >
      <label
        class="sc-EHOje gYSLvW"
        for="color"
      >
        Color
      </label>
      <select
        class="sc-bZQynM lkeYjL"
        id="color"
        name="color"
      >
        <option
          class="sc-gzVnrw enuUcp"
          value="all"
        >
          all  (5)
        </option>
        <option
          class="sc-gzVnrw enuUcp"
          value="black"
        >
          black  (2)
        </option>
        <option
          class="sc-gzVnrw enuUcp"
          value="white"
        >
          white  (3)
        </option>
      </select>
    </div>
    <div
      class="sc-ifAKCX bUenFB"
    >
      <label
        class="sc-EHOje gYSLvW"
        for="character"
      >
        Character
      </label>
      <select
        class="sc-bZQynM lkeYjL"
        id="character"
        name="character"
      >
        <option
          class="sc-gzVnrw enuUcp"
          value="all"
        >
          all  (5)
        </option>
        <option
          class="sc-gzVnrw enuUcp"
          value="modern"
        >
          modern  (2)
        </option>
        <option
          class="sc-gzVnrw enuUcp"
          value="classic"
        >
          classic  (3)
        </option>
      </select>
    </div>
    <div
      class="sc-iwsKbI feoirL"
    >
      <label
        class="sc-gqjmRU kjqYvz"
        for="price"
      >
        Price
      </label>
      <input
        class="sc-gZMcBi cLibKP"
        id="price"
        max="300"
        min="100"
        name="price"
        type="range"
        value="500"
      />
    </div>
    <div
      class="sc-bwzfXH emFNpz"
    >
      <button
        class="sc-htpNat dGEpqy"
      >
        <i
          class="fas fa-th-large"
        />
      </button>
      <button
        class="sc-htpNat fXGbSP"
      >
        <i
          class="fas fa-th"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
